### ISSUE TITLE
Sourcemod (and therefore metamod also) works

### ISSUE THREAD

**Issue by blaberry** [NeotokyoRebuild/neo#836]
Some context on what's required can be found in this chat https://discord.com/channels/335290997317697536/335290997317697536/1319548219755331636

Steps/acceptance criteria
- create a repo under https://github.com/alliedmodders/hl2sdk/branches/all with engine code, maybe mp2013base is enough.
- if it compiles and runs, great
- Create gamedata with signatures (as a compiler step or something?)
- Hook game functions with a test plugin?

**Comment by blaberry** [comment]
I'll refine the steps later, currently boarding a plane.

**Comment by brysondev** [comment]
Getting offsets and sigs should be done for us (at least on linux) by including symbols in the server binary like every other source game. 

Windows requires a bit more leg work and generating byte string sigs, but once we get full party and don't fuck around too much with base classes we should be working as intended.

**Comment by blaberry** [comment]
Yeah from my discussions with people in the SM discord, basically metamod only hooks the virtual function signatures, so I think even if we implement some new classes we might not need to update the SDK repo I linked for every time. But I think the best way to find out for each build is to just test starting a server and seeing if MM/SM works before packaging up the binaries and sending them out. That's a little manual, although it could be automated I guess. But honestly if the steampipe process has a few manual steps or scripts to run it's not the worst imo, not in the short term at least.

**Comment by blaberry** [comment]
We got more context and help from tf2c devs

https://discord.com/channels/335290997317697536/335290997317697536/1319741320880848989

**Comment by blaberry** [comment]
Screenshots 
![Image](https://github.com/user-attachments/assets/b8235ff7-4e15-475b-87c2-bf0f326cfdda)
![Image](https://github.com/user-attachments/assets/7f86e018-d18c-448a-a316-fa192f497b07)

**Comment by brysondev** [comment]
Crossref: #828

@Masterkatze is handling that issue, so I'll assign to them.
